### PR TITLE
Fix URL rewrite for icons in SLD and JSON styles

### PR DESCRIPTION
**Pull request by pvalsecc**
The solution is not 100% perfect (only the URL remapping is taken into account), but it's an easyfix that solves the customers' need.

Fix PNG in SVG layer. Was missing a Batik library for including images in SVG.

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/map/style/SLDParserPlugin.java [hunk 1]
```diff
@@ -112,6 +119,26 @@ private String removeIndexReference(final String styleString) {
...
+                            theUri = theUrl.toURI();
+                        } else {
+                            theUri = URI.create(uri);
+                        }
+                        final ClientHttpRequest request = clientHttpRequestFactory.createRequest(theUri, HttpMethod.GET);
+                        return request.getURI().toURL();
+                    } catch (IOException e) {
+                        return null;
+                    } catch (URISyntaxException e) {
+                        return null;
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/map/style/SLDParserPlugin.java]: The exceptions shouldn't be raised?

File: core/src/main/java/org/mapfish/print/map/style/json/JsonStyleParserHelper.java [hunk 2]
```diff
@@ -119,23 +124,26 @@
     private boolean allowNullSymbolizer;
     private StyleBuilder styleBuilder;
     private Versions version;
+    private final ClientHttpRequestFactory clientHttpRequestFactory;
 
     /**
      * Constructor.
-     *
-     * @param configuration       the configuration to use for resolving relative files or other settings.
+     *  @param configuration       the configuration to use for resolving relative files or other settings.
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/map/style/json/JsonStyleParserHelper.java]: ?

File: core/src/main/java/org/mapfish/print/map/style/json/JsonStyleParserHelper.java [hunk 3]
```diff
@@ -172,6 +180,14 @@ public PointSymbolizer createPointSymbolizer(final PJsonObject styleJson) {
         graphic.graphicalSymbols().clear();
         if (styleJson.has(JSON_EXTERNAL_GRAPHIC)) {
             String externalGraphicUrl = validateURL(styleJson.getString(JSON_EXTERNAL_GRAPHIC));
+            try {
+                final ClientHttpRequest request =
+                        this.clientHttpRequestFactory.createRequest(URI.create(externalGraphicUrl), HttpMethod.GET);
+                externalGraphicUrl = request.getURI().toString();
+            } catch (IOException ignored) {
+                // ignored
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/map/style/json/JsonStyleParserHelper.java]: Why?

**Comment by pvalsecc** [comment]
This PR doesn't lead to working results.

**Pull request closed without merging**